src/PlaywrightRunner.ts

```diff
@@ -52,28 +56,33 @@ const getBrowserTest = (
 const getTests = async (tests: Test[]): Promise<Test[]> => {
   const playwrightPackage = await readPackage()
   const pwTests: Test[] = []
-  await Promise.all(
-    tests.map(async (test) => {
-      const { rootDir } = test.context.config
-      const { browsers, devices } = await readConfig(rootDir)
-      browsers.forEach((browser) => {
-        checkBrowserEnv(browser)
-        const { devices: availableDevices } = getPlaywrightInstance(
-          playwrightPackage,
-          browser,
-        )
-        if (devices && devices.length) {
-          devices.forEach((device) => {
-            const availableDeviceNames = Object.keys(availableDevices)
-            checkDeviceEnv(device, availableDeviceNames)
-            pwTests.push(getBrowserTest(test, browser, device))
-          })
-        } else {
-          pwTests.push(getBrowserTest(test, browser, null))
-        }
-      })
-    }),
-  )
+  const browser2Server: Partial<Record<BrowserType, BrowserServer>> = {}
+  for (const test of tests) {
```

Comment:
there we need for of, because we don't want concurrent/parallel requests when going through them due the singleton of the startup of the browsers won't work otherwise.